docs/admin_api/user_admin_api.rst

```diff
@@ -71,7 +71,7 @@ The parameter ``limit`` is optional but is used for pagination, denoting the
 maximum number of items to return in this call. Defaults to ``100``.
 
 The parameter ``user_id`` is optional and filters to only users with user IDs
-that contain this value.
+or display name that contain this value.
```

Comment:
Mm, actually I talked to some of the other devs and we agree that changing the behaviour of `user_id` from under users isn't the best course of action. Additionally, if someone would like to only filter by `user_id` and not `displayname`, they'd be unable to do so anymore.

Can we add a new field called `user_id_or_displayname` which searches both fields? I agree that it's not the prettiest name but it should do what you desire without implementing complex OR functionality for all fields in the API.

Reply:
Maybe just `name` to search in both? Anyway a problem with searching in the `user_id` still is that it also searches inside the domain part. So if you domain has some very common character sequence you will get a lot of users.

Reply:
> Maybe just `name` to search in both?

This is fine. I'm a little weary that we call the column `name` in the `users` table in Synapse which holds User IDs, but that's just an internal detail. If the docs describe it as both user IDs and displaynames then it should be fine.

> Anyway a problem with searching in the user_id still is that it also searches inside the domain part. So if you domain has some very common character sequence you will get a lot of users.

You could add another field just for displayname, or are you proposing one for `localpart`?

Reply:
My personal opinion is that we would not add separate search fields in the admin UI, so a single query parameter is enough for us.
Searching in both, `user_id` and `displayname` at the same time just gives you a few more results than expected in some cases. It will definitely never be precise enough to select exactly everyone with a specific surname.
For the `user_id` I found a solution to only search inside the `localpart`, see updated PR.